subprojects/packagefiles/taskflow/unittests/meson.build

```diff
@@ -0,0 +1,48 @@
+doctest_dep = dependency('doctest', required: false)
+if doctest_dep.found() and get_option('tests')
+
+if get_option('buildtype') == 'debug' or get_option('buildtype') == 'plain'
+warning('Unittests not effective when buildtype == debug or plain')
+endif
+
+unittest_list = [
+    #'utility',
+    'tsq',
+    #'serializer ',
+    'basics',
+    'asyncs',
+    'subflows',
+    'control_flow',
+    'semaphores',
+    'movable',
+    'cancellation',
+    'algorithms',
+    'compositions',
+    'traversals',
+    'sorting',
+    'pipelines',
+    'scalable_pipelines',
+    'runtimes',
+]
+
+foreach test_group : unittest_list
+
+tf_unittest_exe = executable(test_group,
+    sources: [
+        test_group + '.cpp',
+    ],
+    include_directories: '/usr/include/doctest',
```

Comment:
This should be covered by the doctest dependency, it works when used as a subproject.

Reply:
Acknowledged that it is upstream issue with the CMake dependency module with `doctest`. Thank you for filing the ticket for the Mesonbuild community.

I also applied your suggested changes to make `doctest` dependency required when user specify `-Dtaskflow:tests=true`.

Reply:
@eli-schwartz  Update: I studied the Taskflow codebase again. The author embeds a specific copy of the `doctest.h` in the `3rd-party` directory. So, I edited the meson.build file to use the header-only library by default. This should bypasses the upstream issue https://github.com/doctest/doctest/issues/670 .

I appreciate if you can review the new changes.